src/Domain/CoordinateMaps/Interval.cpp

```diff
@@ -0,0 +1,262 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/Interval.hpp"
+
+#include <cmath>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain::CoordinateMaps {
+
+Interval::Interval(const double A, const double B, const double a,
+                   const double b, const Distribution distribution,
+                   const double singularity_pos) noexcept
+    : A_(A),
+      B_(B),
+      a_(a),
+      b_(b),
+      distribution_(distribution),
+      singularity_pos_(singularity_pos) {
+  ASSERT(
+      A != B and a != b,
+      "The left and right boundaries for both source and target interval must "
+      "differ, but are; ["
+          << A << ", " << B << "] -> [" << a << ", " << b << "]");
+  if (distribution == domain::CoordinateMaps::Distribution::Logarithmic or
+      distribution == domain::CoordinateMaps::Distribution::Inverse) {
+    ASSERT(not std::isnan(singularity_pos),
+           "Both the logarithmic and inverse distribution require "
+           "`singularity_pos` to be specified explicitly.");
+    ASSERT(std::min(a, b) > singularity_pos or std::max(a, b) < singularity_pos,
+           "The singularity for the logarithmic and inverse Interval falls "
+           "inside the domain with ["
+               << std::min(a, b) << ", " << std::max(a, b) << "], but is"
+               << singularity_pos);
+  }
+  is_identity_ =
+      distribution == domain::CoordinateMaps::Distribution::Linear and
+      A == a and B == b;
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> Interval::operator()(
+    const std::array<T, 1>& source_coords) const noexcept {
+  switch (distribution_) {
+    case Distribution::Linear: {
+      return {{((b_ - a_) * source_coords[0] + a_ * B_ - b_ * A_) / (B_ - A_)}};
+    }
+    case Distribution::Equiangular: {
+      return {
+          {0.5 * (a_ + b_ +
+                  (b_ - a_) * tan(M_PI_4 * (2.0 * source_coords[0] - B_ - A_) /
+                                  (B_ - A_)))}};
+    }
+    case Distribution::Logarithmic: {
+      const double logarithmic_zero_ =
```

Comment:
Here are some more variables with a trailing underscore to remove when you get the chance. Trailing underscores are reserved for member variables. Squash directly.